modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -566,52 +423,84 @@ private void dropTableLocally(String name, UUID tblId, List<List<ClusterNode>> a
 
-                    removeListener(TableEvent.CREATE, clo, new IgniteInternalCheckedException(e));
-                }
+                createTblIntention.put(tblId, new CompletableFuture<>());
+
+                clusterCfgMgr
+                    .configurationRegistry()
+                    .getConfiguration(TablesConfiguration.KEY)
+                    .tables()
+                    .change(
+                        change -> change.create(
+                            name,
+                            (ch) -> {
+                                tableInitChange.accept(ch);
+                                ((ExtendedTableChange)ch).
+                                    // Table id specification.
+                                    changeId(tblId.toString()).
+                                    // Affinity assignments calculation.
+                                    changeAssignments(
+                                        ByteUtils.toBytes(
+                                            AffinityService.calculateAssignments(
+                                                baselineMgr.nodes(),
+                                                ch.partitions(),
+                                                ch.replicas()
+                                            )
+                                        )
+                                    ).
+                                    // Table schema preparation.
+                                    changeSchemas(
+                                        schemasCh -> schemasCh.create(
+                                            String.valueOf(INITIAL_SCHEMA_VERSION),
+                                            schemaCh -> schemaCh.changeSchema(
+                                                ByteUtils.toBytes(
+                                                    SchemaService.prepareSchemaDescriptor(
+                                                        ((ExtendedTableView)ch).schemas().size(),
+                                                        ch
+                                                    )
+                                                )
+                                            )
+                                        )
+                                    );
+                            }
+                        )
+                    )
+                    .thenRun(() -> createTblIntention.get(tblId).thenApply(tblFut::complete)
+                        .thenRun(() -> createTblIntention.remove(tblId))
+                        .exceptionally(throwable -> {
+                            createTblIntention.remove(tblId);
+
+                            tblFut.completeExceptionally(new IgniteException(throwable));
+
+                            return null;
+                        }))
+                    .exceptionally(throwable -> {
+                        LOG.error("Table wasn't created [name=" + name + ']', throwable);
```

Comment:
i think it more helpful to use : LoggerMessageHelper.format instead of all these "]" concatenations, wdyt ?

Reply:
Fixed.